contrib/net/http/trace.go

```diff
@@ -41,20 +42,33 @@ type ServeConfig struct {
 // TraceAndServe serves the handler h using the given ResponseWriter and Request, applying tracing
 // according to the specified config.
 func TraceAndServe(h http.Handler, w http.ResponseWriter, r *http.Request, cfg *ServeConfig) {
-	if cfg == nil {
-		cfg = new(ServeConfig)
+	span, ctx := StartRequestSpan(r, cfg.Service, cfg.Resource, cfg.QueryParams, cfg.SpanOpts...)
+	rw, ddrw := wrapResponseWriter(w)
+	defer func() {
+		FinishRequestSpan(span, ddrw.status, cfg.FinishOpts...)
+	}()
+
+	if appsec.Enabled() {
+		h = httpsec.WrapHandler(h, span, cfg.RouteParams)
 	}
```

Comment:
This seems like an important check to keep. Are we risking a nil-pointer exception now?
(If there isn't a test for a nil cfg being passed to this function, then let's add one)

Reply:
My bad, I probably removed it by accident.
Restored by https://github.com/DataDog/dd-trace-go/pull/1286/commits/c2d56a854d2b8e306da303a09b6d95dd79a42d44